modules/internal/DefaultRectangular.chpl

```diff
@@ -644,6 +644,13 @@ module DefaultRectangular {
     pragma "local field"
     var shiftedData : _ddata(eltType);
     var noinit_data: bool = false;
+
+    // 'dataAllocRange' is used by the array-vector operations (e.g. push_back,
+    // pop_back, insert, remove) to allow growing or shrinking the data
+    // buffer in a doubling/halving style.  If it is used, it will be the
+    // actual size of the data buffer, while 'dom' represents the size of the
+    // user-level array.
+    var dataAllocRange: range(int);
```

Comment:
I'd like to leave this for now.  Making it 'bool' would be misrepresenting what the field is for.  Once there is a way to make it 'void' I'll be OK with that.